### PR TITLE
MAINT inspect.getfullargspec is deprecated in Py3.5

### PR DESCRIPTION
**Pull request by ogrisel**
And the code is much nicer with the new signature API.

### CODE REVIEW

File: numpydoc/docscrape.py [hunk 1]
```diff
@@ -460,15 +460,18 @@ def __init__(self, func, role='func', doc=None, config={}):
...
-                argspec = argspec.replace('*', '\*')
-                signature = '%s%s' % (func_name, argspec)
-            except TypeError as e:
+                    # try to read signature, backward compat for older Python
+                    if sys.version_info[0] >= 3:
+                        argspec = inspect.getfullargspec(func)
+                    else:
+                        argspec = inspect.getargspec(func)
+                    argspec = inspect.formatargspec(*argspec)
+                    argspec = argspec.replace('*', '\*')
```
Comment (pv) [inline numpydoc/docscrape.py]: Does inspect.signature deal with this?

Comment (ogrisel) [inline numpydoc/docscrape.py]: hum, probably not indeed. This is for rst escaping right?

Comment (ogrisel) [inline numpydoc/docscrape.py]: I amended my commit to fix this and added a check in the tests.

**Comment by ogrisel** [comment]
The tests pass including the newly introduced check on the escaping of the star character.

**Pull request merged**